### ISSUE TITLE
Actual object 'MR' has no corresponding expected declaration

### ISSUE THREAD

**Issue by detsimov** [icerockdev/moko-resources#271]
When i interop library MLKitTranslate I get an error

```
> Task :shared:linkMultiPlatformLibraryDebugFrameworkIosArm64 FAILED
w: Interop library /Users/detsimov/.gradle/caches/modules-2/files-2.1/dev.icerock.moko/resources-iosarm64/0.17.3/577cde38505e7e10f6eb05196bc2fd429708bb15/resources-cinterop-pluralizedString can't be exported with -Xexport-library
e: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld invocation reported errors
```

**build.gradle.kts**

```
plugins {
    kotlin("multiplatform")
    kotlin("native.cocoapods")
    kotlin("plugin.serialization")
    id("com.android.library")
    id("dev.icerock.mobile.multiplatform.cocoapods")
    id("dev.icerock.mobile.multiplatform.ios-framework")
    id("dev.icerock.mobile.multiplatform-resources")
}

version = "1.0"

kotlin {
    android()
    iosArm64()
    iosX64()

    cocoapods {
        summary = "Some description for the Shared Module"
        homepage = "Link to the Shared Module homepage"
        ios.deploymentTarget = "14.1"
        framework {
            baseName = "MultiPlatformLibrary"
        }
        podfile = project.file("../iosApp/Podfile")
        pod(name = "Firebase/Functions") {
            moduleName = "FirebaseFunctions"
        }
        pod(name = "MLKitTranslate")
    }

    sourceSets {
        val commonMain by getting {
            dependencies {
                api(Libraries.coroutines)
                api(Libraries.koinCore)
                api(Libraries.moko)
                api(Libraries.mokoLiveData)
                api(Libraries.mokoMedia)
                api(Libraries.mokoResources)
                implementation(Libraries.result)
                implementation(Libraries.napier)
                implementation(Libraries.settings)
                implementation(Libraries.settingsNoArg)
                implementation(Libraries.settingsCoroutines)
                implementation(Libraries.settingsSerialization)
                implementation(Libraries.serialization)
                implementation(Libraries.serializationJson)
            }
        }
        val androidMain by getting {
            dependencies {
                implementation(project.dependencies.platform(Libraries.cloudBom))
                implementation(Libraries.cloudTranslate)
                implementation(Libraries.mlKitAndroid)
                implementation(Libraries.ktorAndroid)
                implementation(Libraries.apphud)
                implementation(Libraries.billing)
                implementation(project.dependencies.platform(Libraries.firebaseBom))
                implementation(Libraries.firebaseFunctions)
                implementation(Libraries.firebaseConfig)
                implementation(Libraries.firebaseAnalytics)
                implementation(Libraries.appmetrica)
            }
        }
        val iosArm64Main by getting {
            dependsOn(commonMain)
            dependencies {
            }
        }
        val iosX64Main by getting {
            dependsOn(iosArm64Main)
        }
    }
}

cocoaPods {
    podsProject = file("../iosApp/Pods/Pods.xcodeproj")
}

multiplatformResources {
    multiplatformResourcesPackage = "com.translate.all.language.strings"
    disableStaticFrameworkWarning = true
}

framework {
    export(Libraries.moko)
    export(Libraries.mokoLiveData)
    export(Libraries.mokoResources)
}

android {
    compileSdkVersion(31)
    sourceSets["main"].manifest.srcFile("src/androidMain/AndroidManifest.xml")
    defaultConfig {
        minSdkVersion(21)
        targetSdkVersion(31)
    }
    lintOptions {
        disable("ImpliedQuantity")
    }
}
```

**PodFile**
```

# ignore all warnings from all pods
inhibit_all_warnings!

platform :ios, '14.1'

use_frameworks!
install! 'cocoapods', :disable_input_output_paths => true

pre_install do |installer|
  puts "prebuild..."
  `cd .. && ./gradlew shared:syncMultiPlatformLibraryDebugFrameworkIosX64`
  puts "prebuild completed"
end

target 'Translator' do
  pod 'Firebase/Functions'
  pod 'Firebase/Analytics'
  pod 'Firebase/MLModelDownloader'
  pod 'GoogleMLKit/Translate'
  pod 'MultiPlatformLibrary', :path => '../shared'
  pod 'FlagKit'
  pod 'MultiPlatformLibraryMvvm', :git => 'https://github.com/icerockdev/moko-mvvm.git', :tag => 'release/0.11.0'
end
```

**Podspec**

```
Pod::Spec.new do |spec|
    spec.name                     = 'MultiPlatformLibrary'
    spec.version                  = '0.1.0'
    spec.homepage                 = 'Link to a Kotlin/Native module homepage'
    spec.source                   = { :git => "Not Published", :tag => "Cocoapods/#{spec.name}/#{spec.version}" }
    spec.authors                  = 'IceRock Development'
    spec.license                  = ''
    spec.summary                  = 'Shared code between iOS and Android'

    spec.vendored_frameworks      = "build/cocoapods/framework/#{spec.name}.framework"
    spec.libraries                = "c++"
    spec.module_name              = "#{spec.name}_umbrella"

    spec.pod_target_xcconfig = {
        'MPP_LIBRARY_NAME' => 'MultiPlatformLibrary',
        'GRADLE_TASK[sdk=iphonesimulator*][config=*ebug]' => 'syncMultiPlatformLibraryDebugFrameworkIosX64',
        'GRADLE_TASK[sdk=iphonesimulator*][config=*elease]' => 'syncMultiPlatformLibraryReleaseFrameworkIosX64',
        'GRADLE_TASK[sdk=iphoneos*][config=*ebug]' => 'syncMultiPlatformLibraryDebugFrameworkIosArm64',
        'GRADLE_TASK[sdk=iphoneos*][config=*elease]' => 'syncMultiPlatformLibraryReleaseFrameworkIosArm64'
    }

    spec.script_phases = [
        {
            :name => 'Compile Kotlin/Native',
            :execution_position => :before_compile,
            :shell_path => '/bin/sh',
            #:output_files => ['$TARGET_BUILD_DIR/$PRODUCT_NAME.framework/$PRODUCT_NAME'],
            :script => <<-SCRIPT
                        MPP_PROJECT_ROOT="$SRCROOT/../../shared"
                        MPP_OUTPUT_DIR="$MPP_PROJECT_ROOT/build/cocoapods/framework"
                        MPP_OUTPUT_NAME="$MPP_OUTPUT_DIR/#{spec.name}.framework"
                        "$MPP_PROJECT_ROOT/../gradlew" -p "$MPP_PROJECT_ROOT" "$GRADLE_TASK"
                    SCRIPT
        }
    ]
end
```

**Comment by TSkovsgaard** [comment]
I got this error when upgrading from Kotlin 1.5.31 to 1.6.10, wasn't able to find a solution so ended up rolling back to 1.5.31 for now.

**Comment by mustafaozhan** [comment]
For me, the issue is not with Kotlin version. Moko resources `0.17.3` is failing with this error, while I do not have any issue with `0.17.2`. In both trials, I use Kotlin 1.6.10. It seems like something broken with the latest release.

**Comment by Alex009** [comment]
problem will be fixed at this week

**Comment by Alex009** [comment]
problem from title fixed in #268 (pr #275 )

problem from body - not related to moko-resources. 
```
e: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld invocation reported errors
```
this log tells that something wrong in linking kotlin/native binary with native dependencies (cocoapods that you use). Maybe cocoapods frameworks not found or something else - error should be in log too.

so because problem (from title) related to moko-resources fixed in #268 - i close this issue.
feel free reopen if something related to moko-resources will be found
